snippets/rust.json

```diff
@@ -1,33 +1,446 @@
+        "description": "thread_local!(static …: … = …);"
+    },
+    "try": {
+        "prefix": "try",
+        "body": [
+            "try!(${1})"
+        ],
+        "description": "try!(…)"
     },
     "unimplemented": {
         "prefix": "unimplemented",
         "body": [
             "unimplemented!()"
         ],
-        "description": "Insert unimplemented!"
+        "description": "unimplemented!()"
     },
     "unreachable": {
         "prefix": "unreachable",
         "body": [
-            "unreachable!()"
+            "unreachable!(${1})"
         ],
-        "description": "Insert unreachable!"
+        "description": "unreachable!(…)"
     },
-    "println": {
-        "prefix": "println",
+    "vec": {
+        "prefix": "vec",
         "body": [
-            "println!(\"${1:{\\}}\", $0)"
+            "vec![${1}]"
         ],
-        "description": "Insert println!"
+        "description": "vec![…]"
+    },
+    "write": {
+        "prefix": "write",
+        "body": [
+            "write!(${1}, \"${2}\")"
+        ],
+        "description": "write!(…)"
+    },
+    "writeln": {
+        "prefix": "writeln",
+        "body": [
+            "writeln!(${1}, \"${2}\")"
+        ],
+        "description": "writeln!(…, …)"
+    },
+    "Err": {
+        "prefix": "Err",
+        "body": [
+            "Err(${1})"
+        ],
+        "description": "Err(…)"
+    },
+    "Ok": {
+        "prefix": "Ok",
+        "body": [
+            "Ok(${1:result})"
+        ],
+        "description": "Ok(…)"
+    },
+    "Some": {
+        "prefix": "Some",
+        "body": [
+            "Some(${1})"
+        ],
+        "description": "Some(…)"
+    },
+    "assert": {
+        "prefix": "assert",
+        "body": [
+            "assert!(${1});"
+        ],
+        "description": "assert!(…);"
+    },
+    "assert_eq": {
+        "prefix": "assert_eq",
+        "body": [
+            "assert_eq!(${1}, ${2});"
+        ],
+        "description": "assert_eq!(…, …);"
+    },
+    "bench": {
+        "prefix": "bench",
+        "body": [
+            "#[bench]",
+            "fn ${1:name}(b: &mut test::Bencher) {",
+            "    ${2:b.iter(|| ${3:/* benchmark code */})}",
+            "}"
+        ],
+        "description": "#[bench]"
+    },
+    "const": {
+        "prefix": "const",
+        "body": [
+            "const ${1:CONST}: ${2:Type} = ${4:init};"
+        ],
+        "description": "const …: … = …;"
+    },
+    "derive": {
+        "prefix": "derive",
+        "body": [
+            "#[derive(${1})]"
+        ],
+        "description": "#[derive(…)]"
+    },
+    "else": {
+        "prefix": "else",
+        "body": [
+            "else {",
+            "    ${1:unimplemented!();}",
+            "}"
+        ],
+        "description": "else { … }"
+    },
+    "enum": {
+        "prefix": "enum",
+        "body": [
+            "#[derive(Debug)]",
+            "enum ${1:Name} {",
+            "    ${2:Variant1},",
```

Comment:
Should be just `"    $2"`

Reply:
IFAIK the "Variant1" text is just the default text placed, no ?

Reply:
yes but in some cases it's just a hint for the user

Reply:
and you have just to write hover it